pkg/osv/osv_test.go

```diff
@@ -0,0 +1,50 @@
+package osv
+
+import (
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestRetryOn5xx(t *testing.T) {
+	attempt := 0
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		attempt++
+		w.WriteHeader(http.StatusInternalServerError) // 500
+	}))
+	defer server.Close()
+
+	// Override the QueryEndpoint for testing
+	originalQueryEndpoint := QueryEndpoint
+	QueryEndpoint = server.URL
+	defer func() { QueryEndpoint = originalQueryEndpoint }()
```

Comment:
issue: there's no reason to do this, because `QueryEndpoint` isn't used directly in the function you're testing

(that'll then mean you can make this test a parallel one)